app/customer/routes.py

```diff
@@ -129,18 +136,56 @@ def edit(Id):
             customer.phone_numbers[1].phone_number = addContactForm.other_phone_number.data
 
         # add/edit other_email.
-        if (len(customer.emails) > 1):
-            customer.emails[1].email = addContactForm.other_email.data
         if (len(customer.emails) <= 1 and request.form.get('other_email')):
             addOtherEmail = Email(email=request.form.get('other_email'))
             customer.emails.append(addOtherEmail)
+        if (len(customer.emails) > 1):
+            if customer.emails[1].email != addContactForm.other_email.data:
+                customer.emails[1].email = addContactForm.other_email.data
```

Comment:
I'm probably just not understanding this correctly, but shouldn't this be `customer.emails[#].email`, where # is would be a dynamic number that changes based on how many emails that customer has in the database?  If you're correct here, then I just need help understanding.  Thanks!

Reply:
You're correct, but at the moment the number of emails and phone numbers able to be entered is only 2. From that perspective we know which "other email" in the list that will be edited, always the 2nd. Not sure how to handle more than 2 at the moment, because if we loop through the list like you're saying:

for i in range(len(customer.emails):
         customer.emails[i].email = addContactForm.other_email.data

We wouldn't know which email in the list that the input field from the form is referring to to edit and delete. I suppose a request argument could be sent with the POST request saying which email it is, like "&OtherEmailListIndex=3", but then we'd also have to handle the list index changes when an email gets added or deleted. Also, not sure the most clean way to display more than a couple of emails in the HTML of the customer summary page.

I also forgot to add this conditional statement to the Other Phone Number as well, so pushing another commit now to include it and rebasing.

Reply:
Yeah, I was thinking about how it could be done and I couldn't figure out the best way to do it either.  But yeah, I understand what you're saying and I think we're good now.